csharp/test/Microsoft.ML.OnnxRuntime.Tests/Microsoft.ML.OnnxRuntime.Tests.csproj

```diff
@@ -10,8 +10,49 @@
     <ProtocDirectory>$(OnnxRuntimeBuildDirectory)\$(Configuration)\external\protobuf\cmake\$(Configuration)</ProtocDirectory>
     <ProtoSrc>$(OnnxRuntimeCsharpRoot)\..\onnxruntime\core\protobuf</ProtoSrc>
     <NativeBuildOutputDir>$(OnnxRuntimeBuildDirectory)\$(Configuration)\$(Configuration)</NativeBuildOutputDir>
+
+    <!-- following attributes were necessary for the migrated Tensor tests -->  
+    <LangVersion>7.2</LangVersion>
+    <AllowUnsafeBlocks>True</AllowUnsafeBlocks>
+    <SignAssembly>true</SignAssembly> <!-- need signing for freind access to the internals of the Tensors assembly -->
+    <DelaySign>false</DelaySign>
+    <AssemblyOriginatorKeyFile>..\..\OnnxRuntime.snk</AssemblyOriginatorKeyFile>
+    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
```

Comment:
Why do you need to disable the default compile items? This way you don't need to explicitly list each file in your .csproj.

Reply:
because there is custom compile items with .cs files generated from .tt files.

Reply:
You can solve that following what we do in https://github.com/dotnet/corefxlab/blob/master/src/Microsoft.Data/Microsoft.Data.DataFrame.csproj

For each .tt file:

```xml
    <None Update="BaseColumn.BinaryOperations.tt">
      <Generator>TextTemplatingFileGenerator</Generator>
      <LastGenOutput>BaseColumn.BinaryOperations.cs</LastGenOutput>
    </None>

    <Compile Update="BaseColumn.BinaryOperations.cs">
      <DesignTime>True</DesignTime>
      <AutoGen>True</AutoGen>
      <DependentUpon>BaseColumn.BinaryOperations.tt</DependentUpon>
    </Compile>
```

Just replace `BaseColumn.BinaryOperations` with the name of the .tt and .cs files.